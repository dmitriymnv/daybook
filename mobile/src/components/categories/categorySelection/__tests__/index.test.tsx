import React from 'react';
import renderer from 'react-test-renderer';

import CategorySelection from '../index';

jest.mock('ScrollView', () => {
  const RealComponent = require.requireActual('ScrollView');
  const React = require('React');
  class ScrollView extends React.Component {
    scrollTo() {}

    render() {
      return React.createElement('ScrollView', this.props, this.props.children);
    }
  }
  ScrollView.propTypes = RealComponent.propTypes;
  return ScrollView;
});

it('Должен отобразить кнопки для выбора категории', () => {
  const snap = renderer.create(<CategorySelection />).toJSON();

  expect(snap).toMatchSnapshot();
});
