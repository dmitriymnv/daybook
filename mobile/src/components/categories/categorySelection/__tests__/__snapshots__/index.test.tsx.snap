// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Должен отобразить кнопки для выбора категории 1`] = `
<ScrollView
  horizontal={true}
  showsHorizontalScrollIndicator={false}
  style={
    Object {
      "display": "flex",
      "flexDirection": "row",
      "marginTop": 30,
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#edeef0",
          "borderRadius": 5,
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "center",
          "marginLeft": 15,
          "width": 140,
        }
      }
    >
      <Text />
      <Text
        style={
          Object {
            "lineHeight": 45,
            "marginLeft": 10,
          }
        }
      >
        Бизнес
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#edeef0",
          "borderRadius": 5,
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "center",
          "marginLeft": 15,
          "width": 140,
        }
      }
    >
      <Text />
      <Text
        style={
          Object {
            "lineHeight": 45,
            "marginLeft": 10,
          }
        }
      >
        Игры
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#edeef0",
          "borderRadius": 5,
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "center",
          "marginLeft": 15,
          "width": 140,
        }
      }
    >
      <Text />
      <Text
        style={
          Object {
            "lineHeight": 45,
            "marginLeft": 10,
          }
        }
      >
        Детские
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#edeef0",
          "borderRadius": 5,
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "center",
          "marginLeft": 15,
          "width": 140,
        }
      }
    >
      <Text />
      <Text
        style={
          Object {
            "lineHeight": 45,
            "marginLeft": 10,
          }
        }
      >
        Путешествия
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#edeef0",
          "borderRadius": 5,
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "center",
          "marginLeft": 15,
          "width": 140,
        }
      }
    >
      <Text />
      <Text
        style={
          Object {
            "lineHeight": 45,
            "marginLeft": 10,
          }
        }
      >
        Авто
      </Text>
    </View>
  </View>
</ScrollView>
`;
